### DISCUSSION TITLE
Help on Thread  External Commissioning on 868MHz

### DISCUSSION THREAD

**Discussion by rohithanaydy** [openthread/openthread#7462] [Q&A] [answered]
[Hi] , I am trying to implement Border router and external commissioning in Thread network on 868MHz using EFR32MG13. In my Thread Network RCP is connected to raspberry Pi (Border Router) . I have made some  modifications in the _ot-rcp_ code to make it compatible with 868MHz. Proprietary radio PHY support has been added, Changed some defines (Channel settings) in the _openthread-core-efr32-config.h_
based on the page no 5 in [.]([AN1350: Single-Band Proprietary Sub-GHz Support with OpenThread](https://www.silabs.com/documents/public/application-notes/an1350-openthread-single-band-proprietary-sub-ghz.pdf)) Also in the otbr-posix same changes haves done in those defines in platform.h  under /openthread/blob/main/src/core/config/ folder and built otbr to make 868MHz support. Using the rcp  created the Thread Network with reference to [.]([Thread Network Creation])(https://openthread.io/guides/border-router/external-commissioning/prepare) On-Mesh commissioning is working,  have added 1 router and a child. But I am facing issues in External Commissioning

1. External commissioning is not working
 ~/ot-br-posix/build/otbr/tools
$ ./pskc J01NME dead1111dead2222 SubGHzThread
91aacf8b49e47e203ae6d33064619c8d.
 
 In the Thread Commissioning Android app when I enter this J01NME password to connect to the Border Router ,it shows incorrect password
2. When tried without generating pskc, the default password 123456 also shows as incorrect
3. Using the Open Thread Commissioner app, Selected the Border Router and entered the passphrase. After scanning the QR code then app go back to Border Router Selection page 
Currently using the SL-OPENTHREAD/2.0.0.0_GitHub-9dedd1869 version , is it compatible with these two Commissioning Apps?
4. Some times After set the pskc using the OTBR tool(1), when checking it shows pskc 00000.
5. I have changed the  define OPENTHREAD_CONFIG_PLATFORM_RADIO_PROPRIETARY_CHANNEL_MASK to 0x01fffff, but in the 
 _dataset active_ and _channel supported_ still it shows 0x001ff800(In BR)  I don't  know why it happened ?
6. I know Thread spec have no Sub-GHz support and  GitHub have no Border Router support on 868MHz
  What are the changes to OTBR , making it 868MHz compatible?
7. What changes  I have to make for external commissioning for 868MHz ?

Thanks 
Rohith R

**Comment by gabekassel** [comment]
The OTBR PSKc script is an example, but to my knowledge doesn't actually set the PSKc in the network operational dataset. 

Either as part of creating the initial dataset or as a dataset update after formed, you need to set the PSKc to the derived value you expect. One flow would be:
- `dataset init new`
- `dataset pskc -p YOURPASSWORD`
- `dataset commit active`

https://github.com/openthread/openthread/blob/main/src/cli/README_DATASET.md

I also don't believe the Thread Group app works any longer, though others can confirm, and recommend using the OpenThread app. 

Also, depending what you're testing you can always skip the joiner process and directly set the network key into the end device. This may not be obvious. 

Let us know if this is in the direction of your issue or if you think it's something else.

**Reply by rohithanaydy** [reply]
Thank you Gabekassel for your reply. I  tried the cli command to set the pskc
`sudo ot-ctl pskc -p 123456
sudo ot-ctl dataset init new`

still the Thread Commissioning app is not connected to the Border Router. It shows 123456 as the invalid password.
The Open Thread App also behave like what I told in my doubt.

> Also, depending what you're testing you can always skip the joiner process and directly set the network key into the end device. This may not be obvious.

When I do like this, the devices are automatically added to the Network. I wish to verify the External commissioning in Thread on 868MHz.
 Another doubt is that, I have changed the  OPENTHREAD_CONFIG_PLATFORM_RADIO_PROPRIETARY_CHANNEL_MASK to 0x01fffff in the file   _openthread-core-efr32-config.h_   in the RCP.
But still the  channel mask in dataset shows  0x001ff800

**Comment by rohithanaydy** [comment]
I have changed the OPENTHREAD_CONFIG_PLATFORM_RADIO_PROPRIETARY_CHANNEL_MASK to 0x01fffff in the file openthread-core-efr32-config.h in the RCP.
Also changed OPENTHREAD_CONFIG_PLATFORM_RADIO_PROPRIETARY_CHANNEL_MASK to 0x01fffff  in the platform.h in OTBR under/openthread/blob/main/src/core/config/
But still the channel mask in dataset shows 0x001ff800..
how we can change  the channel mask in OTBR ? 

I need to form the Thread network on the channel < 10

**Reply by jwhui** [reply]
You can use the OpenThread CLI to configure the Operational Dataset's Channel Mask parameter - see [src/cli/README_DATASET.md](https://github.com/openthread/openthread/blob/main/src/cli/README_DATASET.md).

**Comment by rohithanaydy** [comment]
When I set the Channel mask using CLI  [src/cli/README_DATASET.md](https://github.com/openthread/openthread/blob/main/src/cli/README_DATASET.md). Its not working properly. Also The Thread Network is not showing in the Thread Commissioning App.
Even though the channel mask in the Border router is se to 0x01fffff , the channel in the Commissioning app show from 11-26

Thread network prefix in the dataset is different from the Network prefix in the Ip6 of the device.
By doing this when a new device is added to this thread network via On-mesh commissioning, The device added is become leader with the same dataset . Please find the attachment
[email.txt](https://github.com/openthread/openthread/files/8241943/email.txt)

**Comment by schinchani** [comment, marked as answer]
Hi Rohith,

While there could be more than one issue going on here, when trying to work with OT BR on Sub-GHz (868/915 MHz) using the Proprietary PHY option, you will definitely need to rebuild the OpenThread border router application on the host with the following OpenThread stack configurations:

OPENTHREAD_CONFIG_PLATFORM_RADIO_PROPRIETARY_SUPPORT= 1
OPENTHREAD_CONFIG_RADIO_2P4GHZ_OQPSK_SUPPORT=0
OPENTHREAD_CONFIG_RADIO_915MHZ_OQPSK_SUPPORT=0
(Note, that you will need to define **all three configurations** mentioned above)

When these configurations are not set (which is the default case), the radio piece of the stack code built on the host picks the 2.4 GHz option (i.e sets OPENTHREAD_CONFIG_RADIO_2P4GHZ_OQPSK_SUPPORT to 1) in turn causing the channels supported on the proprietary option (0-10) to get masked out. 

One way to do this is by modifying the CMakeList.txt file and specifying the configurations as indicated in this PR:
https://github.com/openthread/ot-br-posix/pull/892

The changes on the RCP side should align with the changes you made for the SoC case to work on 868 MHz. Once you have rebuilt the OTBR application with the above configurations, verify your changes by running the **sudo ot-ctl channel supported** command on your border router. 

Please note that the above instructions are just recommendations. OTBR on 868MHz is not officially tested. Good luck,

Sagar.
